src/Appwrite/Auth/OAuth2/Twitch.php

```diff
@@ -139,11 +137,15 @@ public function getUserName(string $accessToken):string
     protected function getUser(string $accessToken)
     {
         if (empty($this->user)) {
-            $this->user = \json_decode($this->request(
+            $response = \json_decode($this->request(
                 'GET',
-                $this->resourceEndpoint,
-                ['Authorization: Bearer '.\urlencode($accessToken)]
-            ), true)['data']['0'];
+                $this->resourceEndpoint , [
+                    'Authorization: Bearer '.\urlencode($accessToken),
+                    'Client-Id: '. \urlencode($this->appID)
+                ]
+            ), true);
+
+            $this->user = $response['data']['0'];
```

Comment:
```suggestion
            $this->user = $response['data']['0'] ?? [];
```

I would add null safety operator to make sure we don't create any unwanted warnings

Reply:
Okay got it :+1: